server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/cs/factory/DeployedCacheStoreFactory.java

```diff
@@ -18,14 +17,22 @@
  */
 public class DeployedCacheStoreFactory implements CacheStoreFactory {
 
-   private Map<String, DeployedCacheStoreMetadata> deployedCacheStores = Collections.synchronizedMap(new HashMap<String, DeployedCacheStoreMetadata>());
+   private static final int TIMEOUT = 20;
+   private static final TimeUnit TIMEOUT_UNITS = TimeUnit.SECONDS;
+
+   private PromiseMap<String, DeployedCacheStoreMetadata> deployedCacheStores = new PromiseMap<>();
```

Comment:
I encapsulated promise functionality in separate class because backporting to JDK 1.7 would require changing only `PromiseMap` class. However I agree with you... this additional complexity isn't worth it. Let me do the refactoring.